CHANGELOG.md

```diff
@@ -1,13 +1,30 @@
 x.y.z Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
-* None.
+* Redesign `@AsyncOpen` and `@AutoOpen` initialisers to allow setting a client reset modes for any sync configuration,
+ and avoid confusion with the injected configuration and the sync configuration from the user on each intialiser.
 
 ### Fixed
 * <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-swift/issues/????), since v?.?.?)
 * None.
 
-<!-- ### Breaking Changes - ONLY INCLUDE FOR NEW MAJOR version -->
+### Breaking Changes
+* `AsyncOpen.init(appId:partitionValue:configuration:timeout)` and 
+  `AutoOpen.init(appId:partitionValue:configuration:timeout)` will change behavior. Previously 
+  injecting both the configuration and a partition value will make the property wrappers to set
+  the injected configuration (if not nil) combine with the syncConfiguration from the 
+  `user.configuration(partitionValue:)`. This will change to either user the injected configuration (if not nil) 
+  or the configuration obtained from `user.configuration(partitionValue:)` for the current logged user.
+  This API will be deprecated on the next major release.
```

Comment:
Both the old and new behavior seems incorrect? If a non-nil configuration is provided I would expect it to use exactly that configuration except with the partition value changed to the supplied one.